lib/facility.rb

```diff
@@ -1,14 +1,141 @@
+    @address = parse_address(facility_info)
+    @phone = parse_phone(facility_info)
+    @services = parse_services(facility_info)
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def parse_name(data)
+    if data[:name]
+      data[:name]
+    elsif data[:dmv_office]
+      data[:dmv_office]
+    else data[:office_name] && data[:office_type]
+      "#{data[:office_name]}, #{data[:office_type]}"
+    end
+  end
+  
+  def co_address_input?(data) 
+    data[:address_li] && data[:address__1] && data[:city] && data[:state] && data[:zip]
+  end
+
+  def co_address_output(data)
+    "#{data[:address_li]}, #{data[:address__1]}, #{data[:city]}, #{data[:state]} #{data[:zip]}"
+  end
+
+  def co_short_address_input?(data)
+    data[:address_li] && data[:city] && data[:state] && data[:zip]
+  end
+
+  def co_short_address_output(data)
+    "#{data[:address_li]}, #{data[:city]}, #{data[:state]} #{data[:zip]}"
+  end
+
+  def ny_address_input?(data)
+    data[:street_address_line_1] && data[:city] && data[:state] && data[:zip_code]
+  end
+
+  def ny_address_output(data)
+    "#{data[:street_address_line_1]}, #{data[:city]}, #{data[:state]} #{data[:zip_code]}"
+  end
+
+  def mo_address_input?(data)
+    data[:address1] && data[:city] && data[:state] && data[:zipcode]
+  end
+
+  def mo_address_output(data)
+    "#{data[:address1]}, #{data[:city]}, #{data[:state]} #{data[:zipcode]}"
+  end
+
+  def parse_address(data)
+    if data[:address]
+      data[:address] 
+    elsif co_address_input?(data)
+      co_address_output(data)
+    elsif co_short_address_input?(data)
+      co_short_address_output(data)
+    elsif ny_address_input?(data)
+      ny_address_output(data)
+    elsif mo_address_input?(data)
+      mo_address_output(data)
+    end
+  end
+    
+  def parse_phone(data)
+    if data[:phone]
+      data[:phone]
+    else data[:public_phone_number]
+      data[:public_phone_number]
+    end
+  end
+  
+  def parse_services(data)
+    if data[:services_p]
+      data[:services_p]
+    else
+      []
+    end
+  end
+
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    if @services.include?('Vehicle Registration') == true && vehicle.antique? == true
+      vehicle.registration_date = Date.today
+      @collected_fees += 25
+      @registered_vehicles << vehicle
+      vehicle.plate_type = :antique
+    elsif @services.include?('Vehicle Registration') == true && vehicle.electric_vehicle? == true
+      vehicle.registration_date = Date.today
+      @collected_fees += 200
+      @registered_vehicles << vehicle
+      vehicle.plate_type = :ev
+    elsif @services.include?('Vehicle Registration') == true
+      vehicle.registration_date = Date.today
+      @collected_fees += 100
+      @registered_vehicles << vehicle
+      vehicle.plate_type = :regular
+    else @services.include?('Vehicle Registration') == false
+    end
+  end
```

Comment:
This one method is doing a lot of things. It's registering the vehicle AND collecting the fees AND setting the vehicle plate. This goes against SRP and you can consider breaking this method up into separate helper methods:

```ruby
def register_vehicle(vehicle)
  if @services.include?('Vehicle Registration')
    vehicle.register
    self.collect_fees(vehicle)
    self.set_plate_type(vehicle)
    
    return true
  end

  false
end
```

Then you would have 2 helper methods that would collect the fees, and set the plate type based on the engine.

You'll also notice that I dropped all the `== true` since this is unnecessary. Because we are using Boolean logic, the result will either be `true` or `false` there is not need to do the comparison.

Lastly, `else` does not take a conditional so when formatting the `else` branch in your `if` statements, make sure the logic goes underneath and not next to the `else`.

This:

```ruby
else
  return false
end
```

Not this:

```ruby
else @services.include?('Vehicle Registration') == false
end
```